pkg/proxy/logger/endpoint/epinfo.go

```diff
@@ -59,23 +64,35 @@ func (r *endpointInfoRegistry) FillEndpointInfo(info *accesslog.EndpointInfo, ad
 	// Only resolve the security identity if not passed in, as it may have changed since
 	// reported by the proxy. This way we log the security identity and labels used for
 	// policy enforcement, if any.
-	if id == 0 {
+	if info.Identity == 0 {
+		// Try and look up identity by endpoint
 		if ep != nil {
-			id = ep.GetIdentity()
-		} else if addr.IsValid() {
+			secid, _ := ep.GetSecurityIdentity() // ignore error; this is best-effort
```

Comment:
nit: relying on the returned valued being nil/zero when err != nil is generally but not necessarily true.

Reply:
hey man, you got any security IDs? I'm desperate. I'll take anything.


Good point.